src/optimizer/filter_push_down_optimizer.cpp

```diff
@@ -102,7 +102,14 @@ std::shared_ptr<LogicalOperator> FilterPushDownOptimizer::visitCrossProductRepla
     auto buildSchema = op->getChild(1)->getSchema();
     expression_vector predicates;
     std::vector<join_condition_t> joinConditions;
-    for (auto& predicate : remainingPSet.equalityPredicates) {
+
+    JoinType type =
```

Comment:
I wonder if we have a bug where we have equality & inequality predicate, e.g.

```
MATCH (a), (b) WHERE a.x = b.x AND a.age <> b.age
```

Because we will resolve it as Inner HashJoin. But we don't seem to put a.age <> b.age back as a filter